Automattic/sensei
# Fix extra empty header element echoing in course and lesson

Resolves https://github.com/Automattic/themes/issues/7144

## Proposed Changes
* When investigating the spacing issue, we found that there was an extra heading element being added in the lesson content  which was creating issues with the vertical spacing, and it's not specific to the Course theme, it's visible in every theme and adds 8px vertically in most themes, but in themes with block gap, it adds more, for example, in course theme, it adds 20px+8px=28px extra. So apart from fixing it with CSS, we tried also fixing the underlying cause here.

Looking for the source of it, we found that title is echoed using custom template hook (`sensei_single_lesson_content_inside_before`) which calls a function that echoes the title with a wrapping markup inside the lesson template https://github.com/Automattic/sensei/blob/f54f341df5bbb2830652666e08aadf74efdc756e/includes/hooks/template.php#L172.

But now the new issue became why `Sensei_lesson::the_title` function was printing the empty markup without the title, we checked the hooked function here https://github.com/Automattic/sensei/blob/f54f341df5bbb2830652666e08aadf74efdc756e/includes/class-sensei-lesson.php#L4863-L4879, and found that it calls `get_the_title`, and the lesson has a title assigned. But for some reason, it wasn't printing any.

Upon further looking, we found that inside renderer class, there has been another function hooked to the core 'the_title' hook that changes the title to empty https://github.com/Automattic/sensei/blob/4bf75ce9255eea35a6f72aacf3086bc3ff475771/includes/renderers/class-sensei-renderer-single-post.php#L91.

So it makes the title empty, but the markup still prints. To avoid this, and to solve it without causing regression, we added a check for the title before echoing the markup.

## Testing Instructions


1. Disable learning mode
2. Checkout the trunk branch, open a lesson
3. Find an empty header element with class "lesson-title" in the markup.
4. Now checkout this branch
5. Make sure you don't find the above element anymore

![Screenshot 2023-08-07 at 6 28 37 PM](https://github.com/Automattic/sensei/assets/6820724/c6e1f495-347f-409c-a762-f0c2d9b18042)

## Pre-Merge Checklist
